From a7313b8c1690d34ebec3bdc4387a3f590060dbb7 Mon Sep 17 00:00:00 2001
From: Eric Caruso <ejcaruso@chromium.org>
Date: Wed, 7 Dec 2016 16:18:15 -0800
Subject: [PATCH] CHROMIUM: kernel device_jail

Interface to device_jail subsystem found in security/chromiumos.

BUG=chromium:644338

Change-Id: I5aa85b3896b93b19e64825e3626279850c083f77
Signed-off-by: Eric Caruso <ejcaruso@chromium.org>
---
 include/uapi/linux/device_jail.h | 36 ++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 include/uapi/linux/device_jail.h

diff --git a/include/uapi/linux/device_jail.h b/include/uapi/linux/device_jail.h
new file mode 100644
index 000000000000..25ec85555e60
--- /dev/null
+++ b/include/uapi/linux/device_jail.h
@@ -0,0 +1,36 @@
+/*
+ * Device jail user interface.
+ *
+ * Copyright (C) 2016 Google, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#ifndef _UAPI_LINUX_DEVICE_JAIL_H
+#define _UAPI_LINUX_DEVICE_JAIL_H
+
+#include <linux/types.h>
+#include <linux/magic.h>
+
+/* Control device ioctls */
+
+struct jail_control_add_dev {
+	const char __user *path;	/* input */
+	__u32 devnum;			/* output */
+};
+
+#define JAIL_CONTROL_ADD_DEVICE		_IOWR('C', 0, struct jail_control_add_dev)
+#define JAIL_CONTROL_REMOVE_DEVICE	_IOW('C', 1, __u32)
+
+/* Request device responses */
+
+enum jail_request_result {
+	JAIL_REQUEST_ALLOW,
+	JAIL_REQUEST_ALLOW_WITH_LOCKDOWN,
+	JAIL_REQUEST_ALLOW_WITH_DETACH,
+	JAIL_REQUEST_DENY,
+};
+
+#endif /* _UAPI_LINUX_DEVICE_JAIL_H */
-- 
2.22.0.657.g960e92d24f-goog

